src/condor/htcondor_helper.py

```diff
@@ -2,174 +2,332 @@
-                logger.info(f"Cluster {cluster_id} status: {status_msg}. Total wait time: {total_wait} seconds")
-                
-                # for some reason, jobs get 'held' when submitted instead of 'idle'. not sure why.
-                # they do seem to go to 'idle' shortly thereafter though.
-                # 'held' jobs basically also mean 'failed' jobs.
-                # so we will say that if a job is not 'idle' or 'running' after 2 minutes, it has failed.
-                # it's a bit hacky...
-                if total_wait >= 120:
-                    job_is_idle = status_counts.get(1, 0) > 0
-                    job_is_running = status_counts.get(2, 0) > 0
-                    if not job_is_idle and not job_is_running:
-                        raise Exception(f"Cluster {cluster_id} timed out (job was held too long)")
-
-                time.sleep(check_interval)
-                total_wait += check_interval
+                    if not ads:
+                        self.logger.warning(f"No jobs found for cluster {cluster_id}")
+                        return False
+                    
+                    # Check if all jobs completed
+                    completed_jobs = [ad for ad in ads if ad.get("JobStatus") == 4]
+                    if len(completed_jobs) == len(ads):
+                        self.logger.info(f"All jobs in cluster {cluster_id} completed successfully")
+                        return True
+                    
+                    # Check for held jobs
+                    held_jobs = [ad for ad in ads if ad.get("JobStatus") == 5]
+                    if held_jobs:
+                        for job in held_jobs[:3]:  # Log first 3 held jobs
+                            hold_reason = job.get("HoldReason", "Unknown")
+                            hold_code = job.get("HoldReasonCode", 0)
+                            proc_id = job.get("ProcID", "Unknown")
+                            self.logger.warning(f"Job {cluster_id}.{proc_id} held: {hold_reason} (code: {hold_code})")
+                    
+                    # Log status counts
+                    status_counts = {}
+                    for ad in ads:
+                        status = ad.get("JobStatus", 0)
+                        status_counts[status] = status_counts.get(status, 0) + 1
+                    
+                    status_desc = {
+                        1: "Idle", 2: "Running", 3: "Removed",
+                        4: "Completed", 5: "Held", 6: "Transferring Output",
+                        7: "Suspended"
+                    }
+                    
+                    status_msg = ", ".join(f"{status_desc.get(k, f'Unknown({k})')}: {v}" 
+                                           for k, v in status_counts.items())
+                    self.logger.info(f"Cluster {cluster_id} status: {status_msg}. Total wait time: {total_wait} seconds")
+                    
+                    # Retrieve intermediate output files every minute
```

Comment:
feel free to ignore this, but I think fetching intermediate output files is not relevant for this? I think it may be relevant for the command-line version of SKiM-GPT. but for the API I assume it should be all-or-nothing?